### ISSUE TITLE
Section mismatch in sparse_buffer_alloc

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#592]
Reported-by: kernelci.org bot bot@kernelci.org
```
>     WARNING: vmlinux.o(.meminit.text+0x2360): Section mismatch in reference from the function sparse_buffer_alloc() to the function .init.text:sparse_buffer_free()
```

Looks like a straightforward section mismatch; `sparse_buffer_alloc` is NOT marked `__init` but can call `sparse_buffer_free`.  They should match; either both `__init` or not at all.

**Comment by nathanchance** [comment]
Arnd sent a fix: https://lore.kernel.org/lkml/20190709185528.3251709-1-arnd@arndb.de/

**Comment by nickdesaulniers** [comment]
fixed in ae83189405ea5c693683327fa69ac95a23ec59be
